app/index.js

```diff
@@ -166,144 +165,120 @@ module.exports = yeoman.generators.Base.extend({
-    this.gruntfile.insertConfig(
-      'karma',
-      JSON.stringify({
-        unit: {
-          options: {
-            'autoWatch': false,
-            'browsers': this.options.browsers,
-            'configFile': [
-              this.options['config-path'],
-              '/',
-              this.options['config-file']
-            ].join(''),
-            'singleRun': true
+      this.gruntfile.loadNpmTasks('grunt-karma');
+      this.gruntfile.insertConfig(
+        'karma',
+        JSON.stringify({
+          unit: {
+            options: {
+              'autoWatch': false,
+              'browsers': this.options.browsers,
+              'configFile': [
+                this.options['config-path'],
+                '/',
+                this.options['config-file']
+              ].join(''),
+              'singleRun': true
+            }
           }
-        }
-      })
-    );
-    this.gruntfile.registerTask('test', ['karma']);
-  },
-
-  setupTravis: function () {
-    if (!this.options.travis) {
-      return;
-    }
+        })
+      );
+      this.gruntfile.registerTask('test', ['karma']);
+    },
 
-    this.copy('travis.yml', '.travis.yml');
-
-    var done = this.async();
-    var packageJson = path.join(
-      this.options.cwd || process.cwd(),
-      'package.json'
-    );
-
-    fs.readFile(packageJson, { encoding: 'utf-8' }, function (err, content) {
-      var data;
-      if (err) {
-        this.log.error('Could not open package.json for reading.', err);
-        done();
+    setupTravis: function () {
+      if (!this.options.travis) {
         return;
       }
 
-      try {
-        data = JSON.parse(content);
-      } catch (err) {
-        this.log.error('Could not parse package.json.', err);
-        done();
+      this.fs.copy(
+        this.templatePath('travis.yml'),
+        this.destinationPath('.travis.yml')
+      );
+
+      var data = this.fs.readJSON(this.destinationPath('package.json'));
+
+      if (!data) {
+        this.log.error('Could not open package.json for reading.', {});
         return;
       }
 
       if (data.scripts && data.scripts.test) {
-        this.log.writeln(
-          'Test script already present in package.json. Skipping rewriting.'
-        );
-        done();
-        return;
+        this.log.writeln('Test script already present in package.json. Skipping rewriting.');
       }
-
-      data.scripts = data.scripts || {};
-      data.scripts.test = 'grunt test';
-
-      fs.writeFile(packageJson, JSON.stringify(data, null, 2), done);
-    }.bind(this));
+      else{
+        data.scripts = data.scripts || {};
+        data.scripts.test = 'grunt test';
+        this.fs.writeJSON(this.destinationPath('package.json'), data);
+      }
+    }
   },
 
-  installDeps: function () {
-    if (!this.options['skip-install']) {
-        this.npmInstall();
-    }
+  install: function () {
+    this.npmInstall(undefined,undefined,this.async());
```

Comment:
Don't pass any argument here.

Also note that passing `this.async()` callback here will deadlock the node process and just close the generator. - So this is actually breaking the generator :P